src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java

```diff
@@ -163,7 +166,7 @@
     private final Log log;
 
     /** The files to process. */
-    private final Set<File> filesToProcess = new HashSet<>();
+    private final Set<File> filesToProcess = ConcurrentHashMap.newKeySet();
```

Comment:
this is not expected.
Check should work as before in MT mode and completely compatible.
This Check will be good example of how all other Checks in the world(numerous amount of custom/third-party Checks ) continue to work as before.

Reply:
This is a mult-file check and as such it must annotate `GlobalStateful` and have all threads use the same instance. Multi-threads cannot modify the same hashset at the same time. It must have `ConcurrentHashMap`. Check cannot be the same as before since multiple threads are working with it.

Reply:
We modules that cooperate between each other by means of static fields - http://checkstyle.sourceforge.net/config_annotation.html#SuppressWarningsHolder
http://checkstyle.sourceforge.net/config_filters.html#SuppressWarningsFilter

but it will be resolved in https://github.com/checkstyle/checkstyle/issues/4831 , as we resolve this. We should think that it does not exists. This does not really block us, but we will not support such design of Checks. It will work for users with proper management of static fields. Any problems will be user problems.
All modules should be interdependent and should not know anything about other modules.

Reply:
> I did not found any update of our Check to make them thread-safe
This kind of mean that users might not implement Checks that have statics field.

Implementation never got underway or fully implemented, so I wouldn't take no updates as a sign there are no other trouble spots.

> We do not consider right now ability to reuse same Check instance by multiple threads, but we ill discuss this during TreeWalker execution parallelization.

As you have said before, this is only a FileSet issue, so how will this come up in `TreeWalker` discussion?

> We have modules that cooperate between each other by means of static fields 
We should think that it does not exists. 
we will not support such design of Checks/Modules/Filters.
Even they have them, such Checks will not work. Sorry.

Is this documented in our xdocs or somewhere that we don't support these type of modules anymore?
So we supported them before, but no longer. We are fine breaking this and forcing contributors to change their implementation but [not for other types of breaks](https://github.com/checkstyle/checkstyle/issues/5140#issuecomment-396007666)?

Reply:
> so how will this come up in TreeWalker discussion?

ok, lets skip this.

> Is this documented in our xdocs or somewhere that we don't support these type of modules anymore?

Some words are at:
http://checkstyle.sourceforge.net/writingchecks.html#Limitations
`You cannot see the content of other files. You have content of one file only during all Checks execution. All files are processed one by one.`

> So we supported them before, but no longer.

All Holders will work for long time even in MT mode. But we will not guaranty that will keep it forever. As we(probably only I) keep in mind that we are single file validation tool. We do have multi-file, but for now we do not support them as first class entities, they are hacks and till we make a decision on them they are in unclear status. We could drop them, or make them proper API to execute.
Similar decision was on TypeAwareChecks. For now we do not support them, and one day completely remove. And may be in future , we will bring it back with better support.